packages/babel-register/src/node.js

```diff
@@ -61,16 +63,19 @@ function compile(code, filename) {
     }
   }
 
-  const result = babel.transform(
-    code,
-    Object.assign(opts, {
-      // Do not process config files since has already been done with the OptionManager
-      // calls above and would introduce duplicates.
-      babelrc: false,
-      sourceMaps: "both",
-      ast: false,
-    }),
-  );
+  const extendedOpts = Object.assign(opts, {
+    // Do not process config files since has already been done with the OptionManager
+    // calls above and would introduce duplicates.
+    babelrc: false,
+    sourceMaps: "sourceMaps" in opts ? opts.sourceMaps : "both",
+    ast: false,
+  });
+
+  if (extendedOpts.sourceMaps) {
```

Comment:
I think separating the installation from the insertion of `result.map` is somewhat confusing since you're relying on the inputs to Babel instead of the outputs now. What was the motivation for moving this, vs where it was in earlier implementations? I'd probably say

Reply:
Yes, makes sense. Actually, I did that to run the install function before babel transform, which wasn't necessary.